### DISCUSSION TITLE
Custom repository injection without naming convention

### DISCUSSION THREAD

**Discussion by EricMcRay** [mikro-orm/nestjs#33] [Ideas] [answered]
**Is your feature request related to a problem? Please describe.**
I am replacing typeorm with mikro-orm for a project. I replaced all decorators and repositories but just one repository didn't work. After my 4 hour struggle, I realized that I have a typo in the repository name. 

**Describe the solution you'd like**
I am not nestjs pro but inject and resolve repositories from `@Entity` decorators `customRepository` or `EntityRepositoryType` symbol would be nice.

**Describe alternatives you've considered**
Get repository from entity manager 

**Additional context**
And in the custom repository examples [document](https://mikro-orm.io/docs/repositories/#custom-repository) entity and repository names not like nestjs required naming convention so this can be misleading

**Comment by B4nan** [comment]
I literally have no idea what you are asking for. Please provide some examples and be more verbose. 

But don't expect MikroORM decorators or options to have any effect on how nest DI work, that won't be an option, I can say that right ahead.

**Comment by EricMcRay** [comment]
basicly; getting rid of `getRepositoryToken()` mentioned in https://mikro-orm.io/docs/usage-with-nestjs#using-custom-repositories.

**Comment by B4nan** [comment]
How do you expect that to work? All services in nest DI need to have a name. This has nothing to do with MikroORM or its integration, this is how nest works.

**Comment by B4nan** [comment, marked as answer]
not sure if this is what you meant, but as part of supporting nest v8 the repositories are registered based on the references from `@Entity` decorator's `customRepository` option

https://github.com/mikro-orm/nestjs/commit/e51206762f9eb3e96bfc9edbb6abbf7ae8bc08a8#diff-5bb322ab5c33c404e493148fd18a55c0ac6addd1085f0d65a682da1285835909R84

**Reply by EricMcRay** [reply]
this is what I wanted to mean. Sorry for the confision.
